### PR TITLE
doc crypto: add word on using required `CFLAGS`

### PR DESCRIPTION
**Pull request by BytesGalore**

### CODE REVIEW

File: sys/crypto/doc.txt [hunk 1]
```diff
@@ -25,6 +25,12 @@
  * While you can use the ciphers functions directly, you should resort to
  * the generic API for block ciphers whenever possible.
  *
+ * Additionally you need to set a CFLAG for each cipher you want to use in your Makefile:
+ *  * AES-128: CFLAGS += -DCRYPTO_AES
+ *  * 3DES:    CFLAGS += -DCRYPTO_THREEDES
+ *  * Twofish: CFLAGS += -CRYPTO_TWOFISH
+ * This sets the appropriate buffer sizes for cipher operations.
```
Comment (OlegHahm) [inline sys/crypto/doc.txt]: Can you elaborate what exactly gets initialized by using these CFLAGS?

Comment (BytesGalore) [inline sys/crypto/doc.txt]: you mean say that it sets the buffer size of the `cipher_context_t` used by the ciphers for en-/de-cryption operations

Comment (BytesGalore) [inline sys/crypto/doc.txt]: elaborated a bit more and added a missing `D`

File: sys/crypto/doc.txt [hunk 2]
```diff
@@ -25,6 +25,12 @@
  * While you can use the ciphers functions directly, you should resort to
  * the generic API for block ciphers whenever possible.
  *
+ * Additionally you need to set a CFLAG for each cipher you want to use in your Makefile:
+ *  * AES-128: CFLAGS += -DCRYPTO_AES
+ *  * 3DES:    CFLAGS += -DCRYPTO_THREEDES
+ *  * Twofish: CFLAGS += -DCRYPTO_TWOFISH
+ * Setting the CFLAGS initializes a sufficient large buffer size of the cipher_context_t used by the ciphers for en-/de-cryption operations
```
Comment (OlegHahm) [inline sys/crypto/doc.txt]: That's perfect contentwise, but could you please add a linebreak and a fulstop at the end?

**Comment by OlegHahm** [comment]
ACK if the last comment has been addressed.

**Comment by Kijewski** [comment]
Shouldn't we alter [ciphers.h](https://github.com/RIOT-OS/RIOT/blob/7b10e59ef3ef37cde032ddcc4c25b74f781d7428/sys/include/crypto/ciphers.h) to use `#ifdef MODULE_FOO` instead of `#ifdef CRYPTO_THREEDES` instead? The current system is very user-unfriendly even if the documentation is fixed.

**Comment by OlegHahm** [comment]
agreed.

**Comment by BytesGalore** [comment]
@Kijewski would be nice, but we have only the `crypto` module without a hint which cipher is actually used.

**Comment by Kijewski** [comment]
@BytesGalore, ah, every cipher is in the same module. Maybe make the individual algorithms pseudo-modules that depend on module cipher, e.g. `USE_MODULE += cipher_aes128`?

**Comment by BytesGalore** [comment]
@Kijewski sounds reasonable

**Comment by Kijewski** [comment]
Hm, `cipher_context_t` confuses me more than it probably should.

I guess this structure could be redone like
```c
typedef union {
#ifdef MODULE_CIPHER_3DES
    threedes_context_t threedes;
#endif
#ifdef MODULE_CIPHER_AES128
    aes128_context_t aes128;
#endif
#ifdef MODULE_CIPHER_CHACHA
    chacha_context_t chacha;
#endif
#ifdef MODULE_CIPHER_TWOFISH
    twofish_context_t twofish;
#endif
#ifdef MODULE_CIPHER_RC5
    rc5_context_t rc5;
#endif
    char dummy;
} cipher_context_t;
```

(Also I just remembered that I opened #3195 a while back.)

**Comment by Kijewski** [comment]
Maybe merge #4787 in the current state and discuss everything else in a new issue? This PR is an improvement for the meantime.

**Comment by BytesGalore** [comment]
@Kijewski +1 to discuss further in a new issue I opened #4798

**Comment by cgundogan** [comment]
travis is green. GO

**Pull request merged**